lighthouse-cli/chrome-launcher.js

```diff
@@ -0,0 +1,187 @@
+'use strict';
+
+const childProcess = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const mkdirp = require('mkdirp');
+const inquirer = require('inquirer');
+const net = require('net');
+const rimraf = require('rimraf');
+
+const spawn = childProcess.spawn;
+const execSync = childProcess.execSync;
+
+const platformError = new Error('Platform ' + process.platform + ' is not supported');
+
+function defaults(val, def) {
+  return typeof val === 'undefined' ? def : val;
+}
+
+module.exports = class Launcher {
+  constructor(opts) {
+    opts = opts || {};
+    // choose the first one (default)
+    this.head = defaults(opts.head, true);
+
+    this.pollInterval = 500;
+
+    this.chromeInstances = [];
+
+    // prepare tmp dir and chrome outfile and errfile
+    this.prepare();
+
+    // this requires prepare() to be run before
+    // as it depends on TMP_PROFILE_DIR
+    this.flags = [
+      '--remote-debugging-port=9222',
+      '--no-first-run',
+      `--user-data-dir=${this.TMP_PROFILE_DIR}`
+    ];
+  }
+
+  prepare() {
+    switch (process.platform) {
+      case 'darwin':
+      case 'linux':
+        this.TMP_PROFILE_DIR = execSync('mktemp -d -t lighthouse.XXXXXXX').toString().trim();
+        break;
+
+      case 'win32':
+        const tmpdir = process.env.TEMP ||
+          process.env.TMP ||
+          (process.env.SystemRoot || process.env.windir) + '\\temp';
+        const randomNumber = Math.floor(Math.random() * 9e7 + 1e7);
+
+        this.TMP_PROFILE_DIR = path.join(tmpdir, 'lighthouse' + randomNumber);
+        mkdirp.sync(this.TMP_PROFILE_DIR);
+        break;
+
+      default:
+        throw platformError;
+    }
+
+    this.outFile = fs.openSync(`${this.TMP_PROFILE_DIR}/chrome-out.log`, 'a');
```

Comment:
(logging why Chrome is failing is good, I'm just saying it doesn't seem like you'd be able to get the logs if it does fail)